libs/community/langchain_google_community/rank/rank.py

```diff
@@ -0,0 +1,162 @@
+from typing import Any, Optional, Sequence
+
+from google.api_core import exceptions as core_exceptions
+from google.auth.credentials import Credentials  # type: ignore
+from google.cloud import discoveryengine_v1alpha
+from langchain_core.callbacks import Callbacks
+from langchain_core.documents import Document
+from langchain_core.documents.compressor import BaseDocumentCompressor
+from langchain_core.pydantic_v1 import Extra, Field
+
+from langchain_google_community.rank._sdk_manager import VertexRankSDKManager
+
+
+class VertexAIRank(BaseDocumentCompressor):
+    """
+    Initializes the Vertex AI Ranker with configurable parameters.
+
+    Inherits from BaseDocumentCompressor for document processing
+    and validation features, respectively.
+
+    Attributes:
+        project_id (str): Google Cloud project ID
+        location_id (str): Location ID for the ranking service.
+        ranking_config (str):
+            Required. The  name of the rank service config, such as default_config.
+            It is set to default_config by default if unspecified.
+        model (str):
+            The identifier of the model to use. It is one of:
+
+            - ``semantic-ranker-512@latest``: Semantic ranking model
+              with maximum input token size 512.
+
+            It is set to ``semantic-ranker-512@latest`` by default if unspecified.
+        top_n (int):
+            The number of results to return. If this is
+            unset or no bigger than zero, returns all
+            results.
+        ignore_record_details_in_response (bool):
+            If true, the response will contain only
+            record ID and score. By default, it is false,
+            the response will contain record details.
+        title_field (Optional[str]): Specifies the document metadata field
+        to use as title.
+        credentials (Optional[Credentials]): Google Cloud credentials object.
+        credentials_path (Optional[str]): Path to the Google Cloud service
+        account credentials file.
+    """
+
+    project_id: str = Field(default=None)
+    location_id: str = Field(default="global")
+    ranking_config: str = Field(default="default_config")
+    model: str = Field(default="semantic-ranker-512@latest")
+    top_n: int = Field(default=10)
+    ignore_record_details_in_response: bool = Field(default=False)
+    title_field: Optional[str] = Field(default=None)
+    credentials: Optional[Credentials] = Field(default=None)
+    credentials_path: Optional[str] = Field(default=None)
+    sdk_manager: VertexRankSDKManager = Field(default=None)
```

Comment:
I don't think we need an additional `sdk_manager` provided. can we just re-use project_id / location_id attributes from this class instead?